Add unit/e2e tests for mongo-typeorm sample

kiwikern opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe: Docs / Samples
```

## What is the current behavior?
Issue Number: #1539

## What is the new behavior?
Added unit + e2e tests for mongo+typeorm sample.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

BrunnerLivio commented on sample/13-mongo-typeorm/test/photos.e2e-spec.ts:
```diff
@@ -0,0 +1,30 @@
+  let app: INestApplication;
+
+  beforeAll(async () => {
+    const moduleFixture = await Test.createTestingModule({
```
To be consistent
```suggestion
    const module = await Test.createTestingModule({
```

kamilmysliwiec commented:
> Update the test:e2e:samples task to make it work with docker-compose out of the box?

How would that look like?

BrunnerLivio commented:
@kamilmysliwiec Check whether the sample has a `docker-compose.yml` file in the root directory and run `docker-compose up -d` if so.

kiwikern commented:
Thanks for your feedback @BrunnerLivio. 🙂 I addressed your comments and rebased to master. As expected, the e2e test is now failing since the database is not started.

andrewwongyedpay commented on sample/13-mongo-typeorm/src/photo/photo.service.spec.ts:
```diff
@@ -0,0 +1,32 @@
+
+  beforeAll(async () => {
+    const module = await Test.createTestingModule({
+      controllers: [PhotoService],
```
PhotoService in controllers?

kiwikern replied:
Thanks for your feedback. 🙏 I moved the service into the providers array.

kamilmysliwiec commented:
Since the only thing the controller does is delegating work to the injected service instance (which we mock in this test), it doesn't really bring any value (does not protect us from anything).

kiwikern commented:
@kamilmysliwiec I absolutely agree that these tests do not ensure the example code does not break. However, I believe the goal of the tests is to rather serve as examples on how to set up unit/e2e tests for the mongo-sample given. Especially, how you can mock a repository can be of interest for new users.

From the linked issue #1539:
> Give new users like me simple and practical examples for implementing tests
